### PR TITLE
AE2 Cable facade as cover

### PR DESCRIPTION
**Pull request by Glease**
This is tbh not the best code I've ever written, but at least it works. Please comment and review. Suggestions are greatly welcomed.

### CODE REVIEW

**Comment by YannickMG** [comment]
One newly added feature that doesn't properly support this yet is cover tabs, as it's not getting the right kind of facade:
image

**Comment by YannickMG** [comment]
Machine casings are likely to be a popular usage of this feature but they aren't receiving their MACHINE_METAL dye coloration, understandably enough.

**Comment by YannickMG** [comment]
Chisel facades are pretty buggy when removed unfortunately, but seemingly only chisel facades?

#### Textures break entirely on fluid pipes (not item pipes)
removing-cover-pipe

#### Textures just shift to meta 5 for other blocks
removing-cover-other

File: src/main/java/gregtech/api/world/GT_OreData.java [hunk 1]
```diff
@@ -0,0 +1,156 @@
+package gregtech.api.world;
```
Comment (YannickMG) [inline src/main/java/gregtech/api/world/GT_OreData.java:1]: Was `src/main/java/gregtech/api/world/GT_OreData.java` meant to be included in this PR?

Seems unrelated for the time being.

**Comment by YannickMG** [comment]
Looks overall but I didn't look at everything in detail.

File: src/main/java/gregtech/api/world/GT_OreData.java (hunk 1 above)
Comment (Glease) [reply src/main/java/gregtech/api/world/GT_OreData.java:1]: This probably shouldn't be there..

**Comment by draknyte1** [comment]
> Machine casings are likely to be a popular usage of this feature but they aren't receiving their MACHINE_METAL dye coloration, understandably enough.
> 
> image

They’ve existed for a long time already.

File: src/main/java/gregtech/api/net/GT_Packet_RequestCoverData.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+        aOut.writeInt(mZ);
+
+        aOut.writeByte(side);
+        aOut.writeInt(coverID);
+    }
+
+    @Override
+    public GT_Packet_New decode(ByteArrayDataInput aData) {
+        return new GT_Packet_RequestCoverData(
+                aData.readInt(),
```
Comment (repo-alt) [inline src/main/java/gregtech/api/net/GT_Packet_RequestCoverData.java:71]: Does Java actually guarantee parameter evaluation order?

Comment (Glease) [reply src/main/java/gregtech/api/net/GT_Packet_RequestCoverData.java:71]: Yes.

File: src/main/java/gregtech/common/blocks/GT_Block_Casings4.java [hunk 3]
```diff
@@ -67,6 +69,9 @@ public GT_Block_Casings4() {
         ItemList.Casing_EngineIntake.set(new ItemStack(this, 1, 13));
         ItemList.Casing_MiningOsmiridium.set(new ItemStack(this, 1, 14));
         ItemList.Casing_Firebricks.set(new ItemStack(this, 1, 15));
+
+        GT_Mod.gregtechproxy.mCTMBlockCache.put(this, (byte) 6, true);
```
Comment (repo-alt) [inline src/main/java/gregtech/common/blocks/GT_Block_Casings4.java:73]: What is special about these blocks?

Comment (Glease) [reply src/main/java/gregtech/common/blocks/GT_Block_Casings4.java:73]: Fusion casing have connected textures, and thus should have it set to true. It's probably not a very good idea to name this thing cache though...

File: src/main/java/gregtech/common/blocks/GT_Block_Machines.java [hunk 4]
```diff
@@ -648,4 +652,51 @@ public boolean recolourBlock(World aWorld, int aX, int aY, int aZ, ForgeDirectio
...
+                        return facadeBlock;
+                    }
+                }
+            }
+        }
+        return Blocks.air;
+    }
+
+    @Override
+    public int getFacadeMetadata(IBlockAccess aWorld, int aX, int aY, int aZ, int side) {
```
Comment (repo-alt) [inline src/main/java/gregtech/common/blocks/GT_Block_Machines.java:680]: Why not extract common part of these 2 functions

**Review by repo-alt — APPROVED** [review]
Well, I'm unfamiliar with GT texture handling, so can't say much about that part, but otherwise it looks good code wise. (Needless to say that content-wise this is an exellent addition)
